contracts/src/v0.8/dev/interfaces/OCR2DRRegistryInterface.sol

```diff
@@ -0,0 +1,175 @@
+
+  /**
+   * @notice Get configuration relevant for making requests
+   * @return uint32 global max for request gas limit
+   * @return address[] list of registered DONs
+   */
+  function getRequestConfig() external view returns (uint32, address[] memory);
+
+  /**
+   * @notice Determine the charged fee that will be paid to the Registry owner
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @return fee Cost in Juels (1e18) of LINK
+   */
+  function getRequiredFee(bytes calldata data, OCR2DRRegistryInterface.RequestBilling calldata billing)
+    external
+    returns (uint96);
+
+  /**
+   * @notice Estimate the execution cost in gas that will be reimbursed to the Node Operator who transmits the data on-chain
+   * @param billing The request's billing configuration
+   * @return gasAmount
+   */
+  function estimateExecutionGas(OCR2DRRegistryInterface.RequestBilling calldata billing) external returns (uint256);
+
+  /**
+   * @notice Estimate the total cost to make a request: gas re-imbursement, plus DON fee, plus Registry fee
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing The request's billing configuration
+   * @param donRequiredFee Fee charged by the DON that is paid to Oracle Node
+   * @return billedCost Cost in Juels (1e18) of LINK
+   */
+  function estimateCost(
+    bytes calldata data,
+    OCR2DRRegistryInterface.RequestBilling calldata billing,
+    uint96 donRequiredFee
+  ) external view returns (uint96);
+
+  /**
+   * @notice Initiate the billing process for an OCR2DR request
+   * @param data Encoded OCR2DR request data, use OCR2DRClient API to encode a request
+   * @param billing Billing configuration for the request
+   * @return requestId - A unique identifier of the request. Can be used to match a request to a response in fulfillRequest.
+   * @dev Only callable by OCR2DROracles that have been approved on the Registry
+   */
+  function beginBilling(bytes calldata data, RequestBilling calldata billing) external returns (bytes32);
+
+  /**
+   * @notice Finalize billing process for an OCR2DR request
+   * @param requestId contains the proof and response
+   * @param response response data from DON consensus
+   * @param err error from DON consensus
+   * @param transmitter the Oracle who sent the report
+   * @param signers the Oracles who signed the report
+   * @param initialGas the initial amount of gas that was sent by the transmitter when submitting the report
+   * @dev Only callable by OCR2DROracles that have been approved on the Registry
+   * @dev simulated offchain to determine if sufficient balance is present to fulfill the request
+   */
+  function concludeBilling(
+    bytes32 requestId,
+    bytes calldata response,
+    bytes calldata err,
+    address transmitter,
+    address[31] memory signers, // Matches maxNumOracles from OCR2Abstract.sol
```

Comment:
Should this array length be hardcoded?  If so, maybe make it a const variable at the top of the contract ie: `const MAX_NUM_OF_NODES_IN_DON = 31`

Reply:
Awkward part is that you can't have variables in Interfaces. I think I'm just going to remove it for now as it is if/until we split fee payments.